lib/bitstyles_phoenix/alpine3/sidebar.ex

```diff
@@ -0,0 +1,260 @@
+    <svg xmlns="http://www.w3.org/2000/svg" hidden aria-hidden="true">
+      <symbol id="icon-cross" viewBox="0 0 100 100">
+        <path d="M84.12,76.7,57.43,50,84.12,23.3a5.25,5.25,0,0,0-7.42-7.42L50,42.57,23.3,15.88a5.25,5.25,0,0,0-7.42,7.42L42.57,50,15.88,76.7a5.25,5.25,0,1,0,7.42,7.42L50,57.43,76.7,84.12a5.25,5.25,0,1,0,7.42-7.42Z"/>
+      </symbol>
+      <symbol id="icon-hamburger" viewBox="0 0 100 100">
+        <path d="M6,25.57a4.58,4.58,0,0,1,3.23-7.82H88.9a4.58,4.58,0,0,1,0,9.15H9.25A4.54,4.54,0,0,1,6,25.57ZM88.9,45.43H9A4.59,4.59,0,0,0,4.68,50a4.56,4.56,0,0,0,4.57,4.57H88.9A4.58,4.58,0,0,0,93.48,50h0A4.59,4.59,0,0,0,88.9,45.43Zm0,27.67H9.25a4.58,4.58,0,0,0,0,9.15H88.9a4.58,4.58,0,1,0,0-9.15Z"/>
+      </symbol>
+    </svg>
+    """
+  )
+
+  @doc """
+  Renders a sidebar layout with Alpine3.
+
+  Supports all attributes and slots from `BitstylesPhoenix.Component.Dropdown.ui_sidebar`.
+
+  ## Attributes
+  - `x_name` - The name of the boolean x-data property for alpine to store the menu state.
+    Defaults to `sidebarOpen`.
+
+  The `small_sidebar` and `main`/`inner` blocks will additionally provide a block argument
+  with the sidebar context for `ui_js_sidebar_open/1` and `ui_js_sidebar_close/1`.
+
+  ## Prevent page-flickering
+
+  The sidebar small layout sets the `x-cloak` property on the menu to avoid flickering on initial page load.
+  To make this work, you need to add the following snippet to your stylesheets:
+
+     [x-cloak] { display: none !important; }
+
+  See https://alpinejs.dev/directives/cloak for more information on `x-cloak`.
+  """
+
+  def ui_js_sidebar_layout(assigns) do
+    {_, small_extra} = assigns_from_single_slot(assigns, :small_sidebar, optional: true)
+    {_, main_extra} = assigns_from_single_slot(assigns, :main, optional: true)
+
+    extra = assigns_to_attributes(assigns, [:x_data, :small_sidebar])
+    small_extra = Keyword.put_new_lazy(small_extra, :id, &random_id/0)
+
+    assigns =
+      assign(assigns,
+        small_extra: small_extra,
+        extra: extra,
+        main_extra: main_extra,
+        x_data: Map.get(assigns, :x_data, "sidebarOpen")
+      )
+
+    ~H"""
+    <RawSidebar.ui_sidebar_layout x-data={"{ #{@x_data}: false }"} {@extra} >
+      <:small_sidebar
+        x-cloak
+        x-show="sidebarOpen"
+        @click.away="sidebarOpen = false"
+        x-transition:enter="is-transitioning"
+        x-transition:enter-start="is-off-screen"
+        x-transition:enter-end="is-on-screen"
+        x-transition:leave="is-transitioning"
+        x-transition:leave-start="is-on-screen"
+        x-transition:leave-end="is-off-screen"
+        {@small_extra}>
+        <%= assigns[:small_sidebar] && render_block(assigns[:small_sidebar], {@x_data, @small_extra[:id]}) %>
+      </:small_sidebar>
+      <:main {main_extra}>
+        <%= render_block(assigns[:main] || @inner_block, {@x_data, @small_extra[:id]}) %>
+      </:main>
+    </RawSidebar.ui_sidebar_layout>
+    """
+  end
```

Comment:
i don't understand where `large_sidebar` is gone!?

Reply:
It is just passed on to the `RawSidebar` via `extra`